server/external/rapidapiClient.js

```diff
@@ -1,8 +1,78 @@
 const axios = require("axios").default;
+const { checkCache, getCachedCall, cacheData } = require('../redis.repository');
+
+
+const getFromCache = async (requestUrl) => {
+	try {
+        const dataString = await getCachedCall(requestUrl);
+        if (dataString !== null){
+            const data = JSON.parse(dataString);
+            return data;
+        }
+        else { 
+            console.log('expected data, but found none');
+            return null;
+        }
+	} catch (error) {
+	    console.log(error)
+	    return error
+	}
+}
+
+const pushToRedis = async (requestUrl, data) => {
+  // convert response object into JSON string, cache data in redis
+  const dataString = JSON.stringify(data);
+  return await cacheData(requestUrl, dataString);
+}
+
+
+/**
+ * This function completes a request by checking redis to see if cached response for API call exists.
+ * If it exists, this function returns the data as JSON. 
+ * If it does not exist, this function will make the API call, then cache the result in redis for future use.
+ * @param {String} requestUrl 
+ * @param {Object} options 
+ * @returns {Object} will always return the required data, either from cache or from API call
+ */
+const makeQuery = async (requestUrl, options) => {
+    // check for existence of key
+    try {
+        const keyExists = await checkCache(requestUrl);
+        if (keyExists === 1){
+            console.log('key exists in cache, checking now')
+            try {
+                const data = await getFromCache (requestUrl);
+                return data;
+            } catch (error) {
+                console.log(error);
+                return error;
+            }
+        } 
+        // data not in cache
+        else {
+            console.log('key not in cache, calling API');
+            try {
+                // make API call, then cache result for future use
+                console.log('making call')
+                const response = await axios.request(options);
+                const pushstatus = await pushToRedis(requestUrl, response.data);
```

Comment:
Looks good! This will save us from excessive calls to the API during development that may be costly.